### PR TITLE
Add v5 migration guide and update changelog

### PR DESCRIPTION
**Pull request by MarkVillacampa**

### CODE REVIEW

File: Sources/DocCDocumentation/DocCDocumentation.docc/V5_API_Migration_guide.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+### Original Application Version
+
+If you're converting a paid app to in-app subscriptions, and want to provide existing customers with certain features, [we recommend](https://www.revenuecat.com/blog/engineering/converting-a-paid-ios-app-to-subscriptions/) using the "original app version" field in customer info.
+
+Because of a limitation of StoreKit 2, this field is not available if the customer is running on iOS 15, tvOS 15, macOS 12 or watchOS 8. If this is a requirement for you, we recommend switching back to StoreKit 1.
+
+## Trusted Entitlements
+
+Version 5.0 of the SDK enables the Informational mode for Trusted Entitlements by default to protect your purchases from attackers. See the [Trusted Entitlements documentation](https://www.revenuecat.com/docs/trusted-entitlements) for more information.
```
Comment (NachoSoto) [inline Sources/DocCDocumentation/DocCDocumentation.docc/V5_API_Migration_guide.md:70]: I would rephrase this a bit since by default it doesn't actually "protect" you unless you check anything.
Maybe mention that informational will just log verification errors and allow you to check `customer info.entitlements.verification`.

File: Sources/DocCDocumentation/DocCDocumentation.docc/V5_API_Migration_guide.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+The minimum targets have been raised to the folllowing:
+- iOS 13.0
+- tvOS 13.0
+- watchOS 6.2
+- macOS 10.15
+
+## Release Assets
+
+Pre-built .frameworks will no longer be included in releases, only .xcframeworks: https://github.com/RevenueCat/purchases-ios/pull/3582
```
Comment (NachoSoto) [inline Sources/DocCDocumentation/DocCDocumentation.docc/V5_API_Migration_guide.md:82]: Maybe 
```suggestion
Pre-built `.frameworks` are no longer included in releases, only `.xcframeworks`: https://github.com/RevenueCat/purchases-ios/pull/3582
```

File: fastlane/Fastfile [hunk 3]
```diff
@@ -18,7 +18,8 @@ files_with_version_number = {
   './Sources/Misc/SystemInfo.swift' => ['return "{x}"'],
   './.version' => ['{x}'],
   './scripts/docs/index.html' => ['purchases-ios-docs/{x}/documentation/revenuecat'],
-  './scripts/docs/v4_api_migration_guide.html' => ['purchases-ios-docs/{x}/documentation/revenuecat/v4_api_migration_guide']
+  './scripts/docs/v4_api_migration_guide.html' => ['purchases-ios-docs/{x}/documentation/revenuecat/v4_api_migration_guide'],
```
Comment (NachoSoto) [inline fastlane/Fastfile:21]: I think we should remove this one?

Comment (MarkVillacampa) [reply fastlane/Fastfile:21]: Not sure what you mean here. Remove the v4 migration guide?

Comment (NachoSoto) [reply fastlane/Fastfile:21]: Not the file, but this line in `Fastfile`. We don't need to change the version there after 5.x right?

Comment (MarkVillacampa) [reply fastlane/Fastfile:21]: I’m adding the v5 docs, and keeping the v4 docs, the only change in that line is adding a trailing comma. Or am I missing something?

**Review by joshdholtz — APPROVED** [review]
